File: ruby/reverse_string.rb
# def reverse_string(str)
#   # type your code in here
#   reversed = ''
#   length = str.length - 1

#   length.downto(0).each do |i|
#     reversed << str[i]
#   end
#   puts reversed
# end
def reverse_string(str)
    # type your code in here
    reversed = ''
    length = str.length - 1
  
    str.chars.each do |i|
      reversed = i + reversed
    end
     reversed
  end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution

File: javascript/reverse_string.js
function reverseString(str) {
  // type your code here 
   let rev = ''
   for(let i of str){
    rev = i + rev
   }
   return rev

}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));
}

module.exports = reverseString;

// Please add your pseudocode to this file
// And a written explanation of your solution


// for loop,
// set result to empty string, for loop, 